src/ips/algorithm.rs

```diff
@@ -77,30 +80,64 @@ impl Ips {
 
     /// Generate peer list - main function with The Algorithm
     pub async fn generate(&mut self, state: &CrunchyState) -> Vec<Peer> {
+        // Set up logging
+        let output = match self.config.log_path {
+            Some(ref path) => File::create(path).map(|f| Box::new(f) as Box<dyn Write>),
+            None => Ok(Box::new(io::stdout()) as Box<dyn Write>),
+        };
+
+        let mut output = match output {
+            Ok(output) => output,
```

Comment:
If interested, see if we can replace this match with the [unwrap_or_else](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else) fn